### PR TITLE
Minor if-else tweaking

### PR DESCRIPTION
**Pull request by abhimanyu003**
Minor `if-else` tweaking, there is no need to `else`, if there is already `return` in `if` statement.

### CODE REVIEW

File: system/src/Grav/Common/Language/Language.php [hunk 1]
```diff
@@ -386,21 +386,19 @@ public function translate($args, array $languages = null, $array_support = false
...
-                    if (count($args) >= 1) {
-                        return vsprintf($translation, $args);
-                    } else {
-                        return $translation;
-                    }
+                if ($translation and count($args) >= 1) {
+                    return vsprintf($translation, $args);
                 }
+
+                return $translation;
```
Comment (flaviocopes) [inline system/src/Grav/Common/Language/Language.php]: Isn't this actually changing the logic?

Comment (abhimanyu003) [inline system/src/Grav/Common/Language/Language.php]: Oh yes, sorry about that. Updated :)

File: system/src/Grav/Common/Language/Language.php [hunk 2]
```diff
@@ -390,7 +390,9 @@ public function translate($args, array $languages = null, $array_support = false
                     return vsprintf($translation, $args);
                 }
 
-                return $translation;
+                if($translation) {
```
Comment (Quy) [inline system/src/Grav/Common/Language/Language.php]: Add a space after `if`.

Comment (abhimanyu003) [inline system/src/Grav/Common/Language/Language.php]: :+1:

File: system/src/Grav/Common/Data/Blueprint.php [hunk 3]
```diff
@@ -213,14 +213,10 @@ protected function dynamicData(array &$field, $property, array &$call)
...
         list($o, $f) = preg_split('/::/', $function, 2);
-        if (!$f) {
-            if (function_exists($o)) {
-                $data = call_user_func_array($o, $params);
-            }
-        } else {
-            if (method_exists($o, $f)) {
-                $data = call_user_func_array(array($o, $f), $params);
-            }
+        if (!$f && function_exists($o)) {
```
Comment (mahagr) [inline system/src/Grav/Common/Data/Blueprint.php]: This changes the meaning of `if (!$f)`. Old code does nothing of function doesn't exist where new core tries to call `method_exists()` with empty `$f`.

File: system/src/Grav/Common/Language/Language.php [hunk 4]
```diff
@@ -472,12 +472,13 @@ public function getTranslation($lang, $key, $array_support = false)
     public function getBrowserLanguages($accept_langs = [])
     {
         if (empty($this->http_accept_language)) {
-            if (empty($accept_langs) && isset($_SERVER['HTTP_ACCEPT_LANGUAGE'])) {
-                $accept_langs = $_SERVER['HTTP_ACCEPT_LANGUAGE'];
-            } else {
+
+            if (empty($accept_langs) == false && isset($_SERVER['HTTP_ACCEPT_LANGUAGE']) == false) {
```
Comment (mahagr) [inline system/src/Grav/Common/Language/Language.php]: Same as

```php
if (!empty($accept_langs) && !isset($_SERVER['HTTP_ACCEPT_LANGUAGE'])) {
```

This code changes the meaning of the code (resulting undefined variable $_SERVER['HTTP_ACCEPT_LANGUAGE']), use *or* instead:

```php
if (!empty($accept_langs) || !isset($_SERVER['HTTP_ACCEPT_LANGUAGE'])) {
```

Comment (rhukster) [inline system/src/Grav/Common/Language/Language.php]: @Quy There are still a couple of issues with logic that @mahagr has pointed out.  We can't merge this until those are sorted.  Cheers.

**Comment by Sylk** [comment]
Update???

**Comment by mahagr** [comment]
Issues in the pull request need to be fixed or I'll never get accepted.

**Comment by rhukster** [comment]
This PR is too far out of date now.

**Comment by mahagr** [comment]
Also I've been fixing code readability issues like this and will continue doing so whenever I take a look into the code.

**Pull request closed without merging**